src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/FeatureFlagNames.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS;
+
+internal class FeatureFlagNames
+{
+    // Use this prefix name for all feature flags where the feature implementations belongs to dotnet project system.
+    private const string BaseName = "ManagedProjectSystem.";
+
+    internal const string EnableNuGetRestoreCycleDetection = BaseName + "EnableNuGetRestoreCycleDetection";
```

Comment:
nit: When the class is internal, you can mark the members as public, since they'll be internal based on the class access modifier. This is a normal convention.

Reply:
When this has come up in the past, the approach I've seen which makes the most sense to me is to decide whether the member would be public or internal if the class was made public, and mark it accordingly.

Reply:
That totally makes sense. However, I haven't really seen that as an adopted convention. Code gets weird if you try to read into intent that may or may not be there, especially code that is years or decades old.